tests/unit/test_ast.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <string.h>
+#include "assert.h"
+#include "../../src/parser/ast.h"
+#include "../../src/query_executor.h"
+
+void test_validate_set_clause() {
+    char *reason = malloc(sizeof(char) * 30);
```

Comment:
I think the reason why _Validate_*_Clause requires a char ** (reason) is because it's supposed to allocate its memory, here you assume reason of size 30 is enough to hold whatever it needs to hold, which is a bad assumption.

I suggest _Validate_*_Clause will allocate reason if it needs to.